### PR TITLE
Allow for formatting of markdown description for artist series

### PR DESCRIPTION
**Pull request by sweir27**
## Problem
We need to be able to input markdown for Artist Series descriptions and have them render properly. While we do have a coffeescript-based markdown -> html converter already in force, it seems that we prefer to be able to return multiple formats from metaphysics and have this layer do the conversion.

[Note: happy to adjust course if we want to handle markdown in force, but converting to HTML here seemed like the cleanest option that followed existing patterns].

## Solution
For types defined in metaphysics, we have an easy way to include handling for different formats. However, `ArtistSeries` is a type stitched in from Gravity. I extended the type in gravity's stitching file as `descriptionFormatted`. The naming was chosen so that it would show up alphabetically next to `description` in the docs.

## Issues
The type of the `format: ` arg appears to be correct from the perspective of the schema (i.e. I'm forced to enter a correct value), but I could not figure out how to automatically convert the inputted val (like `HTML`, `MARKDOWN`, etc.) into the enum format (`"html"`, `"markdown"`, etc.) so I had to repeat the definition. I'm hoping there's an easy way to do this!!

image

### CODE REVIEW

File: src/lib/stitching/gravity/stitching.ts [hunk 1]
```diff
@@ -176,6 +178,25 @@ export const gravityStitchingEnvironment = (
...
+          fragment: gql`
+            ... on ArtistSeries {
+              description
+            }
+          `,
+          resolve: async ({ description }, { format }) => {
+            if (!isExisty(description) || typeof description !== "string")
+              return null
+
+            const formats = {
```
Comment (sweir27) [inline src/lib/stitching/gravity/stitching.ts:191]: I couldn't figure out a way around this even though this [repeats the definition](https://github.com/artsy/metaphysics/blob/master/src/schema/v2/input_fields/format.ts) of a `GraphQLEnumType` defined elsewhere.

Comment (mzikherman) [reply src/lib/stitching/gravity/stitching.ts:191]: I don't think it matters too much (these arguments are unlikely to change?), but theoretically on could import that type and use it directly. Something like:

```
const { type: formatType } = Format
const desiredFormat = type.getValues().find(e => e.name === format).value

return formatMarkdownValue(description, desiredFormat)
```

**Comment by sweir27** [comment]
Updated based on your comment-- was hoping we could just get that "for free" but does feel better to re-use the existing type!

File: src/lib/stitching/gravity/stitching.ts [hunk 2]
```diff
@@ -176,6 +179,32 @@ export const gravityStitchingEnvironment = (
...
+
+            return formatMarkdownValue(description, desiredFormat)
+
+            const formats = {
+              HTML: "html",
+              PLAIN: "plain",
+              MARKDOWN: "markdown",
+            }
+
+            return formatMarkdownValue(description, formats[format])
```
Comment (mzikherman) [inline src/lib/stitching/gravity/stitching.ts:205]: Should this lower section have been deleted?

Comment (sweir27) [reply src/lib/stitching/gravity/stitching.ts:205]: 😐 yes! following up

**Pull request merged**